### PR TITLE
Add ProphecyComparator for comparing objects that need revealing

### PR DESCRIPTION
**Pull request by jon-acker**
I believe this fixes this issue raised by @stof https://github.com/phpspec/prophecy/issues/69

### CODE REVIEW

File: src/Prophecy/Argument/Token/ObjectStateToken.php [hunk 1]
```diff
@@ -11,6 +11,9 @@
 
 namespace Prophecy\Argument\Token;
 
+use Prophecy\Doubler\LazyDouble;
+use Prophecy\Prophecy\ProphecyInterface;
+use Prophecy\Prophet;
```
Comment (stof) [inline src/Prophecy/Argument/Token/ObjectStateToken.php]: why are you adding them ?

Comment (jon-acker) [inline src/Prophecy/Argument/Token/ObjectStateToken.php]: oops, mistakenly added.

File: src/Prophecy/Comparator/ProphecyComparator.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+namespace Prophecy\Comparator;
+
+use Prophecy\Prophecy\ProphecyInterface;
+use SebastianBergmann\Comparator\ObjectComparator;
+
+class ProphecyComparator extends ObjectComparator
+{
+    public function accepts($expected, $actual)
+    {
+
```
Comment (everzet) [inline src/Prophecy/Comparator/ProphecyComparator.php]: Extra space

**Comment by everzet** [comment]
Looks good beyond extra space :)

**Comment by everzet** [comment]
Thank you very much for contribution!

**Pull request merged**